controllers/music.go

```diff
@@ -0,0 +1,110 @@
+package controllers
+
+import (
+	"log"
+	"net/http"
+  "encoding/xml"
+	"fmt"
+	"io"
+  "html"
+
+	"github.com/UniversityRadioYork/2016-site/structs"
+	"github.com/UniversityRadioYork/2016-site/utils"
+	"github.com/UniversityRadioYork/myradio-go"
+)
+
+// MusicController is the controller for the Music page.
+type MusicController struct {
+	Controller
+}
+
+// Post represents an individual post in the RSS feed
+type Post struct {
+	Title       string `xml:"title"`
+	Link        string `xml:"link"`
+  PubDate     string `xml:"pubDate"`
+  Updated     string `xml:"updated"`
+  Content     string `xml:"encoded"`
+}
+
+// NewMusicController returns a new MusicController with the MyRadio session s
+// and configuration context c.
+func NewMusicController(s *myradio.Session, c *structs.Config) *MusicController {
+	return &MusicController{Controller{session: s, config: c}}
+}
+
+// Get handles the HTTP GET request r for the Music page, writing to w.
+func (sc *MusicController) Get(w http.ResponseWriter, r *http.Request) {
+  // URL of the RSS feed to query
+  url := sc.config.Music.RSSFeed
+
+  isError := false
+
+  // Make HTTP GET request
+  resp, err := http.Get(url)
+  if err != nil {
+    fmt.Printf("Failed to query URL: %v\n", err)
+    RenderError(w, sc)
+    return
+  }
+  defer resp.Body.Close()
+
+  // Read the response body
+  xmlData, err := io.ReadAll(resp.Body)
```

Comment:
Might want to look into using the [xml Decoder](https://pkg.go.dev/encoding/xml#Decoder)'s [Decode](https://pkg.go.dev/encoding/xml#Decoder.Decode) instead of reading and unmarshaling separately

Reply:
Is there a performance benefit to that or is it only about readability?
The XML we're expecting is always limited in size so if it's a performance issue it might not be worth the effort.

Reply:
I imagine it's mostly performance since it can decode the XML stream rather than read the whole response into memory, but IMHO the code is marginally cleaner - compare

```go
xmlData, err := io.ReadAll(resp.Body)
if err != nil {
  // ...
}
var response SomeType
err = xmlUnmarshal(xmlData, &response)
if err != nil {
  // ...
}
```

vs

```go
var response SomeType
err := xml.NewDecoder(resp.Body).Decode(&response)
if err != nil {
  // ...
}
```

Your call, I've not got a strong opinion either way.